doc. also: remove 'guifontset'

justinmk opened the pull request:
- options: remove `'guifontset'` 474ff65
  - It is complicated and not commonly used.
  - It is unlikely to be implemented by Nvim GUIs (complicated to parse, specific to Xorg...).
- doc
  - remove a bunch of noise in `autocmd.txt`
  - `:help bisect` describes how to bisect a vimrc. 
  - https://github.com/neovim/neovim/issues/11960

justinmk committed:
doc: remove redundant autocmd list

This "grouped" list is useless, it only gets in the way when searching
for event names.

blueyed commented:
@justinmk 
I've noticed that `buf_open_scratch` is referred to, but appears to be internal only?
https://github.com/neovim/neovim/blob/26199fedca0c90bdfb1b3279de2ad737fd308e52/runtime/doc/api.txt#L620-L621
Appeared in bec406603.

justinmk commented:
> I've noticed that `buf_open_scratch` is referred to, but appears to be internal only?

`@see` may refer to internal stuff. Haven't thought about how to separate it, low priority for now.

blueyed commented:
Could maybe exclude anything without a `nvim_` prefix then?

justinmk commented:
Actually anything without bars (`|foo()|`) could be excluded I guess.

janlazo commented on runtime/doc/vim_diff.txt:
```diff
@@ -50,7 +50,7 @@ the differences.
 - 'nrformats' defaults to "bin,hex"
 - 'ruler' is enabled
-- 'sessionoptions' excludes "options"
+- 'sessionoptions' excludes "options", includes "unix,slash"
```
When was this added?

justinmk replied:
@janlazo 2c1d12d0beda7b359fec94c00746b7206ae8fedd It's just something I noticed while working on https://github.com/neovim/neovim/pull/7836 and AFAICT there is no reason for platform differences here. This actually makes session files more portable, too. LMK if you disagree.

janlazo replied:
I agree and thanks for the making the change.

janlazo replied:
What about `mkview` and `viewoptions`? It supports `unix` and `slash`.

justinmk replied:
Thanks, will update the doc for that also.

voldikss commented:
Ran into this when I was reading doc/api.txt:

https://github.com/neovim/neovim/blob/a2efc9cf8b0fdf14b01156ba424145e1847f789c/runtime/doc/api.txt#L73

neovim -> pynvim

justinmk committed:
options: remove 'guifontset'

Why:
- It is complicated and is used by almost no one.
- It is unlikely to be implemented by Nvim GUIs (complicated to parse,
  specific to Xorg...).

bfredl commented on runtime/doc/options.txt:
```diff
@@ -5783,7 +5776,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 			   *'statusline'* *'stl'* *E540* *E542*
 'statusline' 'stl'	string	(default empty)
-			global or local to window |global-local|
+			global or local to buffer |global-local|
```
shouldn't be changed, "global or local to window" is correct.

justinmk replied:
I can't explain the behavior in https://github.com/neovim/neovim/pull/11902  though. Is it possible to describe 'statusline' behavior in one sentence? Then we can add that here.

bfredl replied:
But it  behaves just like any "global or local to window" option, and furthermore the behavior described in #11902 is even the behavior for `setlocal` for _any_ window option (global-window and window only differ with `setglobal`). This sounds like just another user who has experimented with window options for two minutes and then decided how window options "works" in a total sense and then expect documentation and other behaviors to be changed because their initial conclusion _must_ be correct and everything that contradicts it _must_ we wrong. we cannot make all these people happy. 

I agree that docs can be improved but this should be a _systematic_ effort (like a rewrite/restructuring of the local options section), and not by adding stray sentences here and there to try to appease one or another specific user who already "knows" how local options work.

justinmk replied:
The "stray sentence" was because I've heard that 'statusline' is special somehow. Is it or isn't it?

bfredl replied:
It used to be the _only_ window global-local option, but is no longer (should be 4-5 of them now). That's all that has been "special" about it.

justinmk committed:
doc [ci skip]

- intro.txt: cleanup
- starting.txt: update, revisit
- doc: `:help bisect`
- mbyte.txt: update aliases 1656367b90bd. closes #11960

justinmk committed:
gen_vimdoc.py: logging
